drivers/led/led_pwm.c

```diff
@@ -65,7 +65,7 @@ static int led_pwm_set_brightness(const struct device *dev,
 	dt_led = &config->led[led];
 
 	return pwm_set_pulse_dt(&config->led[led],
-				dt_led->period * value / 100);
+			(uint32_t) ((uint64_t) dt_led->period * value / 100));
```

Comment:
Hey, not quite asking to change it but reasoning about it. Would it not make sense for this specific use case to just do `dt_led->period / 100 * value` and skip the double casting? Thinking that with the numbers you'll get here for LEDs device periods you'll likely never lose precision anyway.

Reply:
I think we are both being overly cautious, me about wasting few cycles, you about keeping precision on something that does not need to be that precise for its purpose. :-) Ah whatever, I know for a fact that people are using this for stuff that is not LEDs and there's a bunch of `git grep 'uint32_t.*(uint64_t)'` in the code already. :-)